Services/UICore/classes/class.ilTemplate.php

```diff
@@ -335,12 +335,20 @@ function get($part = "DEFAULT", $add_error_mess = false,
 	* Set message. Please use ilUtil::sendInfo(), ilUtil::sendSuccess()
 	* and ilUtil::sendFailure()
 	*/
-	function setMessage($a_type, $a_txt, $a_keep = false)
+	public function setMessage($a_type, $a_txt, $a_keep = false)
 	{
 		if (!in_array($a_type, array("info", "success", "failure", "question")) || $a_txt == "")
 		{
 			return;
 		}
+		global $DIC;
+		if ($DIC->http()
+		        ->request()
+		        ->getQueryParams()[ilFileStandardDropzoneInputGUI::ASYNC_FILEUPLOAD]
+		    && !$a_keep) {
+			echo json_encode([ 'success' => ($a_type !== 'failure'), 'message' => $a_txt ]);
+			exit;
+		}
```

Comment:
That's currently the only position "errors" or "successes" go to... I'm with you, It feels very wrong.